src/Consolonia.Designer/ConsolePreview.cs

```diff
@@ -0,0 +1,451 @@
+            (ushort designWidth, ushort designHeight) = GetDesignWidthAndHeight(xaml);
+            ushort cols = Columns;
+            ushort rows = Rows;
+            if (cols == 0)
+                cols = (ushort)(designWidth / (ushort)_charWidth);
+            if (rows == 0)
+                rows = (ushort)(designHeight / (ushort)_charHeight);
+
+            if (_process == null)
+            {
+                string previewHostPath =
+                    typeof(App).Assembly.Location.Replace(".dll", ".exe", StringComparison.OrdinalIgnoreCase);
+                var processStartInfo = new ProcessStartInfo
+                {
+                    FileName = previewHostPath,
+                    Arguments = $"{xamlPath} --buffer {cols} {rows}",
+                    StandardOutputEncoding = Encoding.UTF8,
+                    StandardInputEncoding = Encoding.UTF8,
+                    UseShellExecute = false,
+                    RedirectStandardInput = true,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = false,
+                    CreateNoWindow = true
+                };
+                _process = Process.Start(processStartInfo);
+
+                if (Design.IsDesignMode)
+                {
+                    string? line = _process!.StandardOutput.ReadLine();
+                    if (!string.IsNullOrEmpty(line))
+                    {
+                        Debug.WriteLine("BUFFER RECEIVED");
+                        var buffer = JsonConvert.DeserializeObject<PixelBuffer>(line)!;
+                        Dispatcher.UIThread.Invoke(() => Content = RenderPixelBuffer(buffer));
+                    }
+
+                    _process.Kill();
+                    _process.Dispose();
+                    _process = null;
+                    return;
+                }
+
+                ListenForChanges();
+            }
+            else
+            {
+                // send request to previewHost to load the new xaml file.
+                _process.StandardInput.WriteLine(xamlPath);
+            }
+        }
+
+        private void ListenForChanges()
+        {
+            Task.Run(async () =>
+            {
+                while (true)
+                {
+                    if (_process == null || _process.HasExited)
+                        return;
+
+                    string? line = await _process!.StandardOutput.ReadLineAsync().ConfigureAwait(false);
+                    if (!string.IsNullOrEmpty(line))
+                    {
+                        Debug.WriteLine("BUFFER RECEIVED");
+#pragma warning disable CA1031 // Do not catch general exception types
+                        try
+                        {
+                            var buffer = JsonConvert.DeserializeObject<PixelBuffer>(line)!;
+                            Dispatcher.UIThread.Invoke(() => Content = RenderPixelBuffer(buffer));
+                        }
+                        catch (Exception ex)
```

Comment:
DeserializeObject is expected to throw few particular exceptions only

Reply:
Changed to JsonException which is base class for a variety of Json Serializer exceptions.